NitroxClient/MonoBehaviours/ThrottledBuilder.cs

```diff
@@ -83,14 +74,29 @@ private void ProcessBuildEventsUntilFrameBlocked()
                 }
 
                 isNextEventFrameBlocked = (processedFrameBlockingEvent && buildEvents.NextEventRequiresFreshFrame());
+
+                if (WaitItem != null && Count != 0)
+                {
+                    int prog = Count - buildEvents.Count;
+                    if (prog < 0)
+                    {
+                        prog = Convert.ToInt32(Count / 2 - prog * -1 / 2);
+                    }
+                    else
+                    {
+                        prog = Convert.ToInt32(prog / 2 + Count / 2);
+                    }
+
+                    WaitItem.SetProgress(prog, Count);
+                }
```

Comment:
This is overloading throttled builder a bit as this use-case is only for loading.  Is there a way to expose the pending count as an attribute and keep all of the progress logic in the building's initial sync?

Reply:
I think simply providing an attribute wouldn't work, because the InitialSync is waiting for completed == true. What i could do, is give the ThrottledBuilder an additional EventHandler that provides the progress. The listener can than be attached and deattached like the FinishedCompletedBuildings.